canonicalwebteam/discourse_docs/models.py

```diff
@@ -0,0 +1,301 @@
+import re
+from urllib.parse import urlparse
+
+from jinja2 import Template
+from requests.exceptions import HTTPError
+
+import dateutil.parser
+import humanize
+from bs4 import BeautifulSoup
+from canonicalwebteam.http import CachedSession
+
+# Constants
+DEFAULT_SESSION = CachedSession(fallback_cache_duration=300)
```

Comment:
You can just initialize the session in the constructor now?

Reply:
the model can receive a custom session, maybe if we start adding new models we might be able to rethink where the session lives.